data_dictionary/storage_options.hh

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2021-present ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <string>
+#include <map>
+#include <seastar/core/sstring.hh>
+
+namespace data_dictionary {
+
+struct storage_options {
+    enum class storage_type {
+        NATIVE, S3,
+    };
+
+    storage_type type;
+    sstring bucket;
```

Comment:
The options below only makes sense for S3, right? If so, maybe you should use a union or std::variant or something? Or at least, a comment say this (if you don't care about wasting a few bytes)?

Reply:
Yes, we could use a std::variant, but it would pretty much waste the same amount of space (actually, at least 1 more byte for storing the variant id). If in the future we'll be supporting more storage formats that need an entirely different set of options, it would make sense to rewrite it to a variant.